component/config.go

```diff
@@ -160,20 +173,40 @@ func MustNewType(strType string) Type {
 
 // DataType is a special Type that represents the data types supported by the collector. We currently support
 // collecting metrics, traces and logs, this can expand in the future.
-type DataType = Type
-
-func mustNewDataType(strType string) DataType {
-	return MustNewType(strType)
-}
+type DataType string
```

Comment:
If we're going to rework all of this, it would make sense to me to move data type into a different package, since things other than components can be types.